opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/_load.py

```diff
@@ -31,9 +30,32 @@
 _logger = getLogger(__name__)
 
 
+class _EntryPointDistFinder:
+    def __int__(self):
+        self._mapping = None
+
+    def dist_for(self, entry_point: EntryPoint):
+        dist = getattr(entry_point, "dist", None)
+        if dist:
+            return dist
+
+        if self._mapping is None:
+            self._mapping = {
+                self._key_for(ep): dist
+                for ep in dist.entry_points
```

Comment:
Is there a more elegant way to obtain the `dist` for a specific `entrypoint` rather than generating and storing info on all installed `dist` s?

Reply:
I don't believe so - importlib doesn't do any sort of caching like pkg_resources did, so this is the most performant way to do this lookup.